src/main/java/org/jabref/gui/contentselector/ContentSelectorDialogView.java

```diff
@@ -0,0 +1,125 @@
+    private ButtonType saveButton;
+
+    private ContentSelectorDialogViewModel viewModel;
+
+    private final BasePanel basePanel;
+    private final DialogService dialogService;
+    private final MetaData metaData;
+
+    public ContentSelectorDialogView(JabRefFrame jabRefFrame) {
+        this.setTitle(Localization.lang("Manage content selectors"));
+        this.getDialogPane().setPrefSize(375, 475);
+
+        this.basePanel = jabRefFrame.getCurrentBasePanel();
+        this.dialogService = jabRefFrame.getDialogService();
+        this.metaData = basePanel.getBibDatabaseContext().getMetaData();
+
+        ViewLoader.view(this)
+                .load()
+                .setAsDialogPane(this);
+
+        ControlHelper.setAction(saveButton, getDialogPane(), event -> saveChangesAndClose());
+    }
+
+    @FXML
+    public void initialize() {
+        viewModel = new ContentSelectorDialogViewModel(metaData, basePanel, dialogService);
+
+        initListView(fieldNamesListView, viewModel::getFieldNamesBackingList, (observable, oldValue, newValue) -> onFieldNameSelected(newValue));
+        initListView(keywordsListView, viewModel::getKeywordsBackingList, (observable, oldValue, newValue) -> onKeywordSelected());
+    }
+
+    @FXML
+    private void addNewFieldName() {
+        viewModel.showInputFieldNameDialog();
+    }
+
+    @FXML
+    private void removeFieldName() {
+        getSelectedFieldName().ifPresent(viewModel::showRemoveFieldNameConfirmationDialog);
+    }
+
+    @FXML
+    private void addNewKeyword() {
+        getSelectedFieldName().ifPresent(viewModel::showInputKeywordDialog);
+    }
+
+    @FXML
+    private void removeKeyword() {
+        Optional<String> fieldName = getSelectedFieldName();
+        Optional<String> keywordToRemove = getSelectedKeyword();
+        if (fieldName.isPresent() && keywordToRemove.isPresent()) {
+            viewModel.showRemoveKeywordConfirmationDialog(fieldName.get(), keywordToRemove.get());
+        }
+    }
+
+    private void initListView(ListView<String> listViewToInit, Supplier<ObservableList<String>> backingList, ChangeListener<String> onSelectedListener) {
+        listViewToInit.setItems(backingList.get());
+        listViewToInit.getSelectionModel().selectedItemProperty().addListener(onSelectedListener);
+        listViewToInit.getSelectionModel().select(FIRST_ELEMENT);
+    }
+
+    private void onFieldNameSelected(String newValue) {
+        removeKeywordButton.setDisable(!getSelectedKeyword().isPresent());
+        removeFieldNameButton.setDisable(!getSelectedFieldName().isPresent());
+        addKeywordButton.setDisable(newValue == null);
+        viewModel.populateKeywordsFor((newValue));
+    }
+
+    private void onKeywordSelected() {
+        removeKeywordButton.setDisable(!getSelectedKeyword().isPresent());
+    }
+
+    private Optional<String> getSelectedFieldName() {
+        return Optional.of(fieldNamesListView.getSelectionModel()).map(SelectionModel::getSelectedItem);
```

Comment:
This should also be moved to the viewModel

Reply:
And if you change this to a BoooleanProperty in the viewModel, you can directly bind to it:
removeFieldNameButton.disableProperty().bind(viewModel.isSelectedFieldNamePrsent))

public BooleanProperty isSelectedFieldNamePresent{
return  ! selectedItemProperty().getValue().isEmpty() 
}

Reply:
Fixed 205170c